### PR TITLE
Added region growing and efficient RANSAC implementations

### PR DESCRIPTION
**Pull request by grzac**

### CODE REVIEW

File: Plugins/CGAL/vtkCGAL/vtkCGALEfficientRANSAC.cxx [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+  this->MinPoints = 200;
+
+  this->Probability = 0.05;
+  this->Epsilon = 0.002;
+  this->ClusterEpsilon = 0.01;
+  this->MaxNormalDeviation = 0.9;
+}
+
+// -----------------------------------------------------------------------------
+// Todo.
```
Comment (fstamour) [inline Plugins/CGAL/vtkCGAL/vtkCGALEfficientRANSAC.cxx]: What;'s TODO?

File: Plugins/CGAL/vtkCGAL/vtkCGALEfficientRANSAC.cxx [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+  // Points with normals.
+  vtkCGALUtilities::Pwn_vector0 points;
+
+  if (!vtkCGALUtilities::vtkPolyDataToPointsWithNormals(input, points))
+  {
+    vtkErrorMacro("Failed to convert input mesh. Make sure to supply points normals.");
+    return 0;
+  }
+
+  std::cout << "Efficient RANSAC" << std::endl;
```
Comment (fstamour) [inline Plugins/CGAL/vtkCGAL/vtkCGALEfficientRANSAC.cxx]: Is this required? Is it just for debugging?

File: Plugins/CGAL/vtkCGAL/vtkCGALRegionGrowing.cxx [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+// Todo.
+vtkCGALRegionGrowing::vtkCGALRegionGrowing()
+{
+  this->MaxDistanceToPlane = 1;
+  this->MaxAcceptedAngle = 45;
+  this->MinRegionSize = 5;
+}
+
+// -----------------------------------------------------------------------------
+// Todo.
```
Comment (fstamour) [inline Plugins/CGAL/vtkCGAL/vtkCGALRegionGrowing.cxx]: Same, what's to do?

**Review by fstamour — COMMENTED** [review]
Maybe replace all the `std::cout` by VTK's logging functions?

https://vtk.org/doc/nightly/html/classvtkLogger.html

File: Plugins/CGAL/vtkCGAL/vtkCGALEfficientRANSAC.cxx (hunk 1 above)
Comment (grzac) [reply Plugins/CGAL/vtkCGAL/vtkCGALEfficientRANSAC.cxx]: Description of the method, a.k.a. what is done in the RequestData of this filter

File: Plugins/CGAL/vtkCGAL/vtkCGALEfficientRANSAC.cxx (hunk 2 above)
Comment (grzac) [reply Plugins/CGAL/vtkCGAL/vtkCGALEfficientRANSAC.cxx]: no, good point

File: Plugins/CGAL/vtkCGAL/vtkCGALRegionGrowing.cxx (hunk 3 above)
Comment (grzac) [reply Plugins/CGAL/vtkCGAL/vtkCGALRegionGrowing.cxx]: same here, some description of the steps achieved in the RequestData is needed

**Comment by grzac** [comment]
@fstamour What's the standard way to print info messages in console for SpecifX? I'll replace the cout calls here but i'm not sure the vtkLogger is the way to go.

The printed messages are for the testers in order to help them fine tuning the parameters. In the mean time I replaced couts by vtkWarningMacro. vtkDebugMacro would be a best fit but I remember it being hidden by default

**Pull request merged**